domain/model/state/modelstate.go

```diff
@@ -106,12 +112,220 @@ func (s *ModelState) Delete(ctx context.Context, uuid coremodel.UUID) error {
+		return constraints.Value{}, errors.Capture(err)
+	}
+	return cons.toValue(tags, spaces, zones)
+}
+
+// getModelConstraintsUUID returns the constraint uuid that is active for the
+// model. If model does not have any constraints then an error satisfying
+// [modelerrors.ConstraintsNotFound] is returned.
+func (s *ModelState) getModelConstraintsUUID(
+	ctx context.Context,
+	tx *sqlair.TX,
+) (string, error) {
+	var constraintUUID dbConstraintUUID
+
+	stmt, err := s.Prepare(
+		"SELECT &dbConstraintUUID.* FROM v_model_constraint",
+		constraintUUID,
+	)
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	err = tx.Query(ctx, stmt).Get(&constraintUUID)
+	if errors.Is(err, sql.ErrNoRows) {
+		return "", errors.New(
+			"no constraints set for model",
+		).Add(modelerrors.ConstraintsNotFound)
+	} else if err != nil {
+		return "", errors.Errorf("getting constraint UUID for model: %w", err)
+	}
+
+	return constraintUUID.UUID, nil
+}
+
+// getModelConstraints returns the values set in the constraints table for the
+// current model. If no constraints are currently set
+// for the model an error satisfying [modelerrors.ConstraintsNotFound] will be
+// returned.
+func (s *ModelState) getModelConstraints(
+	ctx context.Context,
+	tx *sqlair.TX,
+) (dbConstraint, error) {
+	stmt, err := s.Prepare("SELECT &dbConstraint.* FROM v_model_constraint", dbConstraint{})
+	if err != nil {
+		return dbConstraint{}, errors.Capture(err)
+	}
+
+	var constraint dbConstraint
+	err = tx.Query(ctx, stmt).Get(&constraint)
+	if errors.Is(err, sql.ErrNoRows) {
+		return dbConstraint{}, errors.New(
+			"no constraints set for model",
+		).Add(modelerrors.ConstraintsNotFound)
+	}
+	if err != nil {
+		return dbConstraint{}, errors.Errorf("getting model constraints: %w", err)
+	}
+	return constraint, nil
+}
+
+// deleteModelConstraints deletes all constraints currently set on the current
+// model. If no constraints are set for the current model or no model exists
+// then no error is raised.
+func (s *ModelState) deleteModelConstraints(
+	ctx context.Context,
+	tx *sqlair.TX,
+) error {
+	constraintUUID, err := s.getModelConstraintsUUID(ctx, tx)
+	if errors.Is(err, modelerrors.ConstraintsNotFound) {
+		return nil
+	} else if err != nil {
+		return errors.Errorf("getting constraints uuid for model: %w", err)
+	}
+
+	stmt, err := s.Prepare(`DELETE FROM model_constraint`)
+	if err != nil {
+		return errors.Capture(err)
+	}
+	err = tx.Query(ctx, stmt).Run()
+	if err != nil {
+		return errors.Errorf("delete constraints %q for model: %w", constraintUUID, err)
+	}
+
+	dbConstraintUUID := dbConstraintUUID{UUID: constraintUUID}
+
+	stmt, err = s.Prepare(
+		"DELETE FROM constraint_tag WHERE constraint_uuid = $dbConstraintUUID.uuid",
+		dbConstraintUUID,
+	)
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	err = tx.Query(ctx, stmt, dbConstraintUUID).Run()
+	if err != nil {
+		return errors.Errorf("deleting model constraint %q tags: %w", constraintUUID, err)
```

Comment:
`deleting model constraint tags for %q` would be easier to read as the messy bit is at the end

Reply:
your error doesn't tell me what the id applies to. The model or constraints? This is also a pretty established pattern I feel. Are we talking about a big change now?

Reply:
It's the same as the error in the PR just with a different word order to make the core message more human readable. We an discuss out of band.